src/js/components/AppListItemLabelsComponent.jsx

```diff
@@ -0,0 +1,159 @@
+var classNames = require("classnames");
+var React = require("react/addons");
+
+var OnClickOutsideMixin = require("react-onclickoutside");
+var Util = require("../helpers/Util");
+
+var AppListItemLabelsComponent = React.createClass({
+  displayName: "AppListItemLabelsComponent",
+
+  initialMarginTop: null,
```

Comment:
OK, I see the _initial_ top margin must be stored somehow.

They are ways to do this.
But doing this in a member variable is not one of these ways. ;)

__First__ approach, you could use the unique data-reactid to store the value in a private object:
```javascript
var id = dropdownNode.attributes["data-reactid"].value;

_initialTopMargins[id] = Math.abs(dropdownNode.offsetTop);
```
This entry has to be removed at some point, while componentWillUnmount ?, to prevent memory leaks.

The __second__ approach would be your suggestion to store this at a data-attribute, but I am not sure if this has side effects, because you are modifying the DOM after react render.. And it should be slower, because of the DOM write access.

__Third__ approach, we introduce the StorageMixin like in DCOS UI, but for now we never had a need for that and it lead to a careless (in react terms) programming style, because you aren't aware of the state so much anymore. It's shifting around the react patterns.